lib/domain/usecase/generate_thumbnails_media_interactor.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/send_file_dialog/generate_thumbnails_media_state.dart';
+import 'package:fluffychat/presentation/extensions/send_file_web_extension.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_file_extension.dart';
+import 'package:matrix/matrix.dart';
+
+typedef OnConvertReadStreamToBytesDone = void Function(
+  MatrixFile oldFile,
+  MatrixFile newFile,
+);
+
+class GenerateThumbnailsMediaInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Room room,
+    required List<MatrixFile> files,
+    required OnConvertReadStreamToBytesDone onConvertReadStreamToBytes,
```

Comment:
Why we need it when we listened stream event?

Reply:
because when matrixFile is converted from readStream to bytes, it's the different matrixFile, so in controller, we need to update these changes to update view also